.github/workflows/validate_plan_environments.yml

```diff
@@ -0,0 +1,46 @@
+name: TF Validate and Plan for all environments
+
+# runs workflow on push and if any changes are detected in any .tf files
+on:
+  workflow_call:
+    inputs:
+      commit-identifier:
+        description: "SHA or tag to validate and plan terraform, (optional) when omitted, the head of the selected branch is used"
+        type: string
+        required: true 
+      terraform-workspace:
+        description: Terraform workspace that is used
+        type: string
+        required: true
```

Comment:
you can wait to workout the details on this, but I think the terraform-workspace may have to be a single value sent in but if left blank, then it defaults the environment.... complications of not using the default workspace and in some repos we use the same workspace name across all environments and in others use use dev/stage/prod...